homeassistant/components/shelly/sensor.py

```diff
@@ -70,6 +86,19 @@ def name(self):
     @property
     def state(self):
         """Value of sensor."""
+        if self.attribute in [
+            "deviceTemp",
+            "extTemp",
+            "humidity",
+            "overpowerValue",
+            "power",
+        ]:
+            return round(getattr(self.block, self.attribute), 1)
+        # Energy unit change from Wmin or Wh to kWh
+        if self.info[aioshelly.BLOCK_VALUE_UNIT] == "Wmin":
+            return round(getattr(self.block, self.attribute) / 60 / 1000, 2)
+        if self.info[aioshelly.BLOCK_VALUE_UNIT] == "Wh":
+            return round(getattr(self.block, self.attribute) / 1000, 2)
```

Comment:
You need to add logic to the constructor to also make sure `self._unit` is set to the correct unit.

Reply:
Done